shared/settings/advanced/index.tsx

```diff
@@ -208,6 +216,136 @@ class Developer extends React.Component<Props, State> {
   }
 }
 
+// A list so the order of the elements is fixed
+const proxyTypeList = ['noProxy', 'httpConnect', 'socks']
+const proxyTypeToDisplayName = {
+  httpConnect: 'HTTP Connect',
+  noProxy: 'No Proxy',
+  socks: 'Socks5',
+}
+
+type ProxyState = {
+  address: string
+  port: string
+  proxyType: string
+}
+
+class ProxySettings extends React.Component<Props, ProxyState> {
+  state = {
+    address: '',
+    port: '',
+    proxyType: 'noProxy',
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.proxyData !== this.props.proxyData) {
+      const addressPort = this.props.proxyData.addressWithPort.split(':')
+      const address = addressPort[0]
+      var port = '80'
+      if (addressPort.length >= 2) {
+        port = addressPort[1]
+      }
+
+      const proxyType = RPCTypes.ProxyType[this.props.proxyData.proxyType]
+      this.setState({address, port, proxyType})
+    }
+  }
+
+  toggleCertPinning = () => {
+    if (this.certPinning()) {
+      this.props.onDisableCertPinning()
+    } else {
+      this.props.onEnableCertPinning()
+    }
+  }
+
+  saveProxySettings = () => {
+    const proxyData = {
+      addressWithPort: this.state.address + ':' + this.state.port,
+      certPinning: this.certPinning(),
+      proxyType: (RPCTypes.ProxyType[this.state.proxyType] as unknown) as RPCTypes.ProxyType,
+    }
+    this.props.saveProxyData(proxyData)
+  }
+
+  certPinning = (): boolean => {
+    if (this.props.allowTlsMitmToggle === null) {
+      if (this.props.proxyData) {
+        return this.props.proxyData.certPinning
+      } else {
+        return true // Default value
+      }
+    } else {
+      return !this.props.allowTlsMitmToggle
+    }
+  }
+
+  proxyTypeSelected = (proxyType: string) => {
+    var cb = () => {}
+    if (proxyType === 'noProxy') {
+      // Setting the proxy type to no proxy collapses the menu including the save button, so save immediately
+      cb = this.saveProxySettings
+    }
+    this.setState({proxyType}, cb)
+  }
+
+  renderProxySettings() {
+    if (this.state.proxyType === 'noProxy') {
+      return null
+    }
+    return (
+      <Kb.Box direction="vertical" alignItems="flex-start" style={styles.expandedProxyContainer}>
```

Comment:
This "alignItems" doesn't do anything, and there's a warning on the console about it -- it's a property used by style={}.